File: 백준/Silver/10828. 스택/스택.py
import sys

n = int(sys.stdin.readline())
stack = []
for i in range(n):
    a = sys.stdin.readline().split()
    # push일때
    if 'push' in a:
        a[1] = int(a[1])
        stack.append(a[1])
    # pop일때
    elif a[0]=='pop':
        if len(stack)==0:
            print(-1)
        else:
            print(stack.pop())
    # size 일때
    elif a[0]=='size':
        print(len(stack))
    # empty일때
    elif a[0]=='empty':
        if len(stack)==0:
            print(1)
        else:
            print(0)
    # top일때
    elif a[0]=='top':
        if len(stack)==0:
            print(-1)
        else:
            print(stack[-1])

File: 백준/Silver/10828. 스택/스택.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        Stack<Integer> stack = new Stack<>();
        int n = Integer.parseInt(br.readLine());
        for (int i = 0; i<n; i++) {
            String a = br.readLine();
            if (a.equals("pop")) {
                if (stack.isEmpty()) {
                    System.out.println(-1);
                } else {
                    System.out.println(stack.pop());
                }
            } else if (a.equals("size")) {
                System.out.println(stack.size());
            } else if (a.equals("empty")) {
                if (stack.isEmpty()) {
                    System.out.println(1);
                } else {
                    System.out.println(0);
                }
            } else if (a.equals("top")) {
                if (stack.isEmpty()) {
                    System.out.println(-1);
                } else {
                    System.out.println(stack.peek());
                }
            } else {
                String[] result = a.split(" ");
                stack.push(Integer.valueOf(result[1]));
            }
        }
    }
}